### ISSUE TITLE
Ballerina client does not send the client cert in a mutual SSL scenario when using the JRE default certs as trusted certs

### ISSUE THREAD

**Issue by Bhashinee** [ballerina-platform/ballerina-standard-library#4890]
**Description:**

Ballerina Client does not send the client certificate in the SSL handshake when you create a Ballerina client configuring JRE default certs as the trusted certs along with the private key and client certs to be used in a mutual SSL handshake. 

The following code does not work with a mutual SSL enabled server.

```ballerina
import ballerina/http;
import ballerina/io;

public function main() returns error? {
    http:Client okapiClient = check new ("https://localhost:9090/albums", {
            secureSocket: {
                key: {
                    certFile: "./resources/client.pem",
                    keyFile: "./resources/client.key"
                }
            }
    });
    json[]|http:ClientError payload = okapiClient->get("/");
    if payload is json {
        io:println(payload);
    } else {
        io:println(payload);
    }     
}

```

Server-side error,

`error:100000c0:SSL routines:OPENSSL_internal:PEER_DID_NOT_RETURN_A_CERTIFICATE`

**Comment by Bhashinee** [comment]
This is due to a bug in the HTTP/2 Ballerina implementation. This works without any issues when used with HTTP/1.1. As a workaround, we can configure the client to use HTTP/1.1 as follows,

import ballerina/http;
import ballerina/io;

public function main() returns error? {
    http:Client okapiClient = check new ("https://localhost:9090/albums", {
            secureSocket: {
                key: {
                    certFile: "/Users/bhashinee/Documents/Development/standard-libraries/module-ballerina-http/ballerina-tests/resources/certsandkeys/public.crt",
                    keyFile: "/Users/bhashinee/Documents/Development/standard-libraries/module-ballerina-http/ballerina-tests/resources/certsandkeys/private.key"
                }
            },
            httpVersion: "1.1"
    });
    json[]|http:ClientError payload = okapiClient->get("/");
    if payload is json {
        io:println(payload);
    } else {
        io:println(payload);
    }     
}

**Comment by Bhashinee** [comment]
This fix is available with HTTP 2.10.2.
